api-references/data/insights.json

```diff
@@ -1,1177 +1,2682 @@
+              "title": "Report ID for which insights are needed",
+              "type": "string"
             },
-            "DepositData": {
-                "title": "DepositData",
-                "required": ["account"],
-                "type": "object",
-                "properties": {
-                    "account": { "$ref": "#/components/schemas/Account" }
-                }
+            "name": "reportId",
+            "in": "query"
+          },
+          {
+            "required": true,
+            "schema": {
+              "title": "x-client-id",
+              "type": "string"
             },
-            "ErrorResponse": {
-                "title": "ErrorResponse",
-                "required": ["errorCode", "errorMsg"],
-                "type": "object",
-                "properties": {
-                    "errorCode": { "title": "Errorcode", "type": "string" },
-                    "errorMsg": { "title": "Errormsg", "type": "string" },
-                    "timestamp": {
-                        "title": "Timestamp",
-                        "type": "string",
-                        "format": "date-time"
-                    },
-                    "ver": { "title": "Ver", "type": "string", "default": "1" },
-                    "txnid": {
-                        "title": "Txnid",
-                        "type": "string",
-                        "default": ""
-                    }
-                }
+            "name": "x-client-id",
+            "in": "header"
+          },
+          {
+            "required": true,
+            "schema": {
+              "title": "x-client-secret",
+              "type": "string"
             },
-            "FIData": {
-                "$ref": "#/components/schemas/DepositData",
-                "title": "FIData"
+            "name": "x-client-secret",
+            "in": "header"
+          },
+          {
+            "required": true,
+            "schema": {
+              "title": "x-product-instance-id",
+              "type": "string"
             },
-            "FIDataBlockCreateResponse": {
-                "title": "FIDataBlockCreateResponse",
-                "required": ["body"],
-                "type": "object",
-                "properties": {
-                    "status": {
-                        "title": "Status",
-                        "type": "string",
-                        "default": "Success"
-                    },
-                    "statusDescription": {
-                        "title": "Statusdescription",
-                        "type": "string"
+            "name": "x-product-instance-id",
+            "in": "header"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful Response",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "title": "Response 200 Get Insights V1 Insight Get",
+                  "anyOf": [
```

Comment:
Any reason to use `anyOf` and not `oneOf`?

Reply:
I think this has been generated by the backend service. This API already exists on docs, I haven't modified it. Do our docs not support `anyOf` today?

Reply:
Understood. Currently, we do `allOf`, `oneOf`, `anyOf` should be similar, I'll get this working. Closing in favour of fixing for implementing `oneOf`.